modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/streaming/PagingProviderProducer.java

```diff
@@ -141,6 +147,7 @@ public int getSize() {
       return function.apply(connection);
     } catch (Exception caughtException) {
       if (isFirstPage) {
+        tryToMutateConfigurationStats(config, (MutableConfigurationStats::discountActiveComponent));
```

Comment:
Can it happen that if we are in the first page, we are actually calling the discountActiveComponent twice?

Reply:
No because if an exception is caught on getting the first page, the whole paging provider is disposed and a new one is created later.